### DISCUSSION TITLE
One-to-many matching

### DISCUSSION THREAD

**Discussion by erelsgl** [networkx/networkx#4686] [Algorithms]
In a bipartite graph (X+Y,E), in a one-to-many matching, each vertex v in X has an integer capacity c[v], and it can be matched to at most c[v] vertices of Y. Each vertex inY may be matched to at most one vertex of X.

A one-to-many matching can be found by reduction to one-to-one matching: constuct c[v] clones of each vertex v in X, and find a standard maximum matching in the resulting graph. 

Would you consider adding such an algorithm?

**Comment by dschult** [comment]
Yes, we would consider such an addition. Please include references to relevant articles.
Thanks...
